### PR TITLE
GraphQL better tests

### PR DESCRIPTION
**Pull request by chakeresa**
**Changes proposed in this pull request:**
 - DRY up tests of GraphQL responses by adding `compare_gql_and_db_dogs`, `compare_gql_and_db_users`, `dog_type_attributes`, and `user_type_attributes` to `rails_helper`.
 - Test edge cases of no user/dog with the given `id` when querying `user` or `dog`

**The following checks have been completed:**
 - [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
 - [x] Checked `coverage/index.html` - did not add any new code that's not covered by testing (if possible)
 - [x] Merged in the latest master to my branch with `git pull origin master` & resolved merge conflicts
 - [x] Ran `rails db:migrate`
 - [x] Ran the test suite - all tests are passing (or maybe skipped)
 - [x] Checked affected endpoints in Postman / GraphiQL
 - [x] Updated README for changes (new endpoints, new gems, etc)

### CODE REVIEW

File: spec/graphql/queries/dog_spec.rb [hunk 1]
```diff
@@ -9,45 +9,26 @@
     json = JSON.parse(response.body)
```
Comment (alexander-mathieu) [inline spec/graphql/queries/dog_spec.rb]: 🏜

File: spec/graphql/queries/dog_spec.rb [hunk 2]
```diff
@@ -9,45 +9,26 @@
...
-      'firstName'   => user.first_name,
-      'lastName'    => user.last_name,
-      'email'       => user.email,
-      'shortDesc'   => user.short_desc,
-      'longDesc'    => user.long_desc
-    )
+    compare_gql_and_db_users(actual_user, user)
+  end
+
+  it 'raises exception if no dog with that id' do
```
Comment (alexander-mathieu) [inline spec/graphql/queries/dog_spec.rb]: This is interesting.  I'm still having a hard time convincing myself that we're testing anything here!

File: spec/graphql/queries/dog_spec.rb [hunk 3]
```diff
@@ -9,45 +9,26 @@
...
-          id
-          name
-          breed
-          weight
-          birthdate
-          shortDesc
-          longDesc
+          #{dog_type_attributes}
           user {
-            id
```
Comment (alexander-mathieu) [inline spec/graphql/queries/dog_spec.rb]: 🏜
